### PR TITLE
feat(ui): add accept screenshot button to the UI mode

### PR DESCRIPTION
**Pull request by shadowusr**
### Overview & Demo
This PR implements an ability to accept screenshots in UI mode.

### What's done
- A tick appears next to a failed screenshot. If you click on it, the actual screenshot is accepted and the button is changed to "undo".
- Save buttons appear next to actual and expected attachments of a failed screenshot. Button is disabled if the click won't change anything (screenshot's already accepted)
- If some screenshots are modified, corresponding tests are marked as modified in the test tree with yellow circle
- A "modified" filter has been added

### Why
This PR partially addresses #24310.
In short, it's highly inconvenient to switch between CLI mode and UI mode to accept screenshots. As of now, you can't interact with screenshots state in UI mode which makes its use very limited

### CODE REVIEW

**Comment by shadowusr** [comment]
@microsoft-github-policy-service agree

**Comment by pavelfeldman** [comment]
The patch is much larger than a typical contribution we get and upon looking at it, I have review comments regarding pretty much every other line, starting with the reporter bits you used to capture the original path all the way to the changes in the UI, leaving alone the UX you chose. I don't think the review will converge, so I'd suggest to either wait for us to implement it or to split it into sensible review-able patches that we would be able to land one by one.

**Comment by shadowusr** [comment]
@pavelfeldman, I'm interested in implementing this feature myself, because we need it now and it's completely unclear when playwright team will be able to implement it.

> I have review comments regarding pretty much every other line

I'd appreciate if you could list at least several high-level comments regarding this PR, what specifically needs to be changed.

> leaving alone the UX you chose

If you don't think the chosen UX is the best choice, let's discuss what issues you see with it and make sure we're on the same page before I dive into further edits.

> split it into sensible review-able patches that we would be able to land one by on

Okay, I can do that, just want to make sure those PRs won't be silently closed as this one and discuss aforementioned points in advance. It takes time, and I don't want it to go to waste.

As for splitting into multiple PRs, we can do it like this:
1. Implement the most basic screenshots accepting experience by adding "save" buttons near the attachments
2. Add "Accept actual" / "Undo" button near the `toHaveScreenshot` command
3. Add test status updating and filtering capabilities

Sounds good to you?

File: docs/src/test-api/class-testinfo.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ Learn more about [test annotations](../test-annotations.md).
   - `contentType` <[string]> Content type of this attachment to properly present in the report, for example `'application/json'` or `'image/png'`.
   - `path` ?<[string]> Optional path on the filesystem to the attached file.
   - `body` ?<[Buffer]> Optional attachment body used instead of a file.
+  - `targetPath` ?<[string]> Path on the filesystem to the original file in project to which this attachment relates to.
```
Comment (aslushnikov) [inline docs/src/test-api/class-testinfo.md:34]: We treat public API as a huge maintenance cost; all changes or additions to the surface have to be justified and approved by all core team members.

In practice, only core functionality and highest pressing concerns result in public API changes. The "Accept screenshot button" is not important enough to justify this change, so this can't go in.

File: packages/playwright-test/src/isomorphic/teleReceiver.ts [hunk 2]
```diff
@@ -526,7 +526,7 @@ export class TeleTestCase implements reporterTypes.TestCase {
 
 export type TeleTestResult = reporterTypes.TestResult & {
   stepMap: Map<string, reporterTypes.TestStep>;
-  statusEx: reporterTypes.TestResult['status'] | 'scheduled' | 'running';
+  statusEx: reporterTypes.TestResult['status'] | 'scheduled' | 'running' | 'modified';
```
Comment (aslushnikov) [inline packages/playwright-test/src/isomorphic/teleReceiver.ts:529]: Telereceiver is used in a few places around; we're not comfortable adding a new execution status for such a niche usecase.

Niche features should be implemented in a well-isolated manner. An easy heuristic that could be used is "could this feature be removed by only deleting lines of code"?

File: packages/playwright-test/src/matchers/toMatchSnapshot.ts [hunk 3]
```diff
@@ -212,19 +212,20 @@ class SnapshotHelper<T extends ImageComparatorOptions> {
     else
       output.push('');
 
+    const targetPath = path.relative(process.cwd(), this.snapshotPath);
```
Comment (aslushnikov) [inline packages/playwright-test/src/matchers/toMatchSnapshot.ts:215]: `process.cwd()` should never be used in matchers (and generally, we prefer to never use is at all).

File: packages/trace-viewer/src/ui/uiModeView.tsx [hunk 4]
```diff
@@ -166,75 +175,77 @@ export const UIModeView: React.FC<{}> = ({
...
-      <div className='vbox'>
-        <div className={'vbox' + (isShowingOutput ? '' : ' hidden')}>
-          <Toolbar>
-            <div className='section-title' style={{ flex: 'none' }}>Output</div>
-            <ToolbarButton icon='circle-slash' title='Clear output' onClick={() => xtermDataSource.clear()}></ToolbarButton>
-            <div className='spacer'></div>
-            <ToolbarButton icon='close' title='Close' onClick={() => setIsShowingOutput(false)}></ToolbarButton>
-          </Toolbar>
-          <XtermWrapper source={xtermDataSource}></XtermWrapper>
+  return <AppContext.Provider value={ { isUIMode: true } }>
```
Comment (aslushnikov) [inline packages/trace-viewer/src/ui/uiModeView.tsx:178]: The UX: accepting the screenshot should happen next to the screenshot diff view since that's where the decision happens.
